src/main/java/no/sikt/nva/pubchannels/channelregistrycache/db/service/CacheService.java

```diff
@@ -44,21 +46,51 @@ public static CacheService defaultInstance() {
   }
 
   public void loadCache(S3Client s3Client) {
-    var entries = ChannelRegistryCsvLoader.load(s3Client).getCacheEntries();
-    var uniqueEntries = filterDuplicatesBasedOnPid(entries);
-    int start = 0;
-    while (start < uniqueEntries.size()) {
-      var writeBatchBuilder =
-          WriteBatch.builder(ChannelRegistryCacheDao.class).mappedTableResource(table);
-      uniqueEntries.subList(start, Math.min(start + BATCH_SIZE, uniqueEntries.size())).stream()
-          .map(ChannelRegistryCacheEntry::toDao)
-          .forEach(writeBatchBuilder::addPutItem);
-      var writeBatch = writeBatchBuilder.build();
-      client.batchWriteItem(r -> r.addWriteBatch(writeBatch));
-      start += BATCH_SIZE;
-      LOGGER.info("Loaded {} entries", start);
+    var loader = new ChannelRegistryCsvLoader(s3Client);
+    var result = loader.getEntries();
+
+    var counter = new AtomicInteger(0);
+    var batchCounter = new AtomicInteger(0);
+    var batch = new ArrayList<ChannelRegistryCacheDao>(BATCH_SIZE);
+    var seenPids = new ConcurrentHashMap<UUID, Boolean>();
+    var batchLock = new ReentrantLock();
+
+    result
+        .entries()
+        .filter(entry -> seenPids.putIfAbsent(entry.getPid(), Boolean.TRUE) == null)
+        .map(ChannelRegistryCacheEntry::toDao)
+        .forEach(
+            dao -> {
+              counter.incrementAndGet();
+              batchLock.lock();
```

Comment:
tried really hard to keep a stream all the way but this lock probably eats up the advantage... well well